lib/i18n/hygiene/checks/missing_interpolation_variable_check.rb

```diff
@@ -0,0 +1,33 @@
+require 'i18n/hygiene/wrapper'
+require 'i18n/hygiene/checks/base'
+require 'i18n/hygiene/variable_checker'
+require 'i18n/hygiene/result'
+
+module I18n
+  module Hygiene
+    module Checks
+      class MissingInterpolationVariableCheck < Base
+        def run
+          puts "Checking for mismatching interpolation variables..."
+
+          wrapper = I18n::Hygiene::Wrapper.new
+
+          mismatched_variables = wrapper.keys_to_check.select do |key|
+            checker = I18n::Hygiene::VariableChecker.new(key, wrapper, config.locales)
+            checker.mismatch_details if checker.mismatched_variables_found?
+          end
+
+          mismatched_variables.each { |details| puts details }
+
+          puts "Finished checking.\n\n"
+
+          if mismatched_variables.any?
+            yield Result.new(:failure)
+          else
+            yield Result.new(:pass)
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
At some point I think it'd be worth examining how responsibilities are shared between the check class (like `Checks::MissingInterpolationVariableCheck`) and the next class down (like `I18n::Hygiene::VariableChecker`).

Do we need both? If so, what should the boundary be? At the moment it seems like the top class is responsible for determining which keys to check, and the bottom class is instantiated once per key.